### PR TITLE
Rearrange tor and tgen configuration files

### LINKED ISSUE #15: Re-organize torrc files for usability
We want to reorganize torrc files similar to how was done here:
https://github.com/shadow/shadow/pull/1411

the `conf/` dir will have:
- tor.common.torrc
  - tor.relay.torrc
    - tor.relay.authority.torrc
    - tor.relay.exitonly.torrc
    - tor.relay.exitguard.torrc
    - tor.relay.guardonly.torrc
    - tor.relay.other.torrc (no exit or guard position flag)
  - tor.client.torrc
    - tor.client.markov.torrc
    - tor.client.perf.torrc

and then the host-specific directories of each process will have a local torrc that includes a subset of the files above according to the characteristics of that Tor process (relay, exit, or client etc).

We think this will make it easier for users to make edits to Tor configs. since they could just change a config for a single Tor node by editing the local torrc in the cwd of the node.

### PR DESCRIPTION
**Pull request by stevenengler**
Closes #15, and also rearrange the tgen markov client configuration files to store them in the hosts directory rather than the conf directory.

### CODE REVIEW

File: tornettools/generate_defaults.py [hunk 1]
```diff
@@ -49,8 +56,7 @@
 
 TGENRC_SERVER_FILENAME="tgen-server.tgenrc.graphml"
 TGENRC_PERFCLIENT_FILENAME="tgen-perf.tgenrc.graphml"
-TGENRC_MARKOVCLIENT_DIRNAME="tgen-markov"
-TGENRC_MARKOVCLIENT_FILENAME_FMT="{}.tgenrc.graphml"
+TGENRC_MARKOVCLIENT_FILENAME="tgenrc.graphml"
 TGENRC_FLOWMODEL_FILENAME_FMT="flowmodel.{}usec.graphml"
```
Comment (robgjansen) [inline tornettools/generate_defaults.py:60]: Nit: should we use this instead to be consistent with the stream and packet model names?

```
TGENRC_FLOWMODEL_FILENAME_FMT="tgen.tor-flowmodel-{}usec.graphml"
```

**Review by robgjansen — APPROVED** [review]
Looks plausible!

It's a bit hard to tell the structure of everything without actually running the code to generate a new config structure. Should I do that as a sanity check?

Also, do you think this change is big enough that we should run a trial Tor experiment to make sure the experiment runs as expected?

File: tornettools/generate_defaults.py (hunk 1 above)
Comment (stevenengler) [reply tornettools/generate_defaults.py]: Done.

**Comment by stevenengler** [comment]
> Also, do you think this change is big enough that we should run a trial Tor experiment to make sure the experiment runs as expected?

I ran a 0.1% network and Shadow completed successfully. I'll run another one before merging and I'll double check that the log files look okay.

**Pull request merged**